api/types/events/events.proto

```diff
@@ -105,6 +105,15 @@ message ConnectionMetadata {
     string Protocol = 3 [ (gogoproto.jsontag) = "proto,omitempty" ];
 }
 
+// ClientMetadata identifies the originating client for an event.
+message ClientMetadata {
+    // UserAgent identifies the type of client that attempted the event.
+    string UserAgent = 1 [ (gogoproto.jsontag) = "user_agent,omitempty" ];
+
+    // IPAddress identifies the IP of the client that attempted the event.
+    string IPAddress = 2 [ (gogoproto.jsontag) = "ip_address,omitempty" ];
```

Comment:
I can't see any harm in also including the remote port here, but not sure if it overly adds any value ? Potentially help line logs up better ?